### ISSUE TITLE
Velero restore partially failed as default priorityClass

### ISSUE THREAD

**Issue by MallyZhang** [vmware-tanzu/velero#5288]
**What steps did you take and what happened:**
Velero restore partially failed as only one priorityClass can be default
```
time="2022-09-06T01:48:25Z" level=info msg="restore status includes excludes: <nil>" logSource="pkg/restore/restore.go:1134" restore=velero/velero-backup-obj-20220906014657-20220906014748
time="2022-09-06T01:48:25Z" level=info msg="Attempting to restore PriorityClass: hc-default" logSource="pkg/restore/restore.go:1245" restore=velero/velero-backup-obj-20220906014657-20220906014748
time="2022-09-06T01:48:25Z" level=error msg="error restoring hc-default: priorityclasses.scheduling.k8s.io \"hc-default\" is forbidden: PriorityClass hc-default is already marked as default. Only one default can exist" logSource="pkg/restore/restore.go:1357" restore=velero/velero-backup-obj-20220906014657-20220906014748
```

```
[root@localhost ]# kubectl create -f hc-default.yaml 
+ kubectl create -f hc-default.yaml
Error from server (Forbidden): error when creating "hc-default.yaml": priorityclasses.scheduling.k8s.io "hc-default" is forbidden: PriorityClass hc-default is already marked as default. Only one default can exist
[root@localhost ]# kubectl create -f hc-db.yaml 
+ kubectl create -f hc-db.yaml
Error from server (AlreadyExists): error when creating "hc-db.yaml": priorityclasses.scheduling.k8s.io "hc-db" already exists
```

**What did you expect to happen:**

don't report this error when there is one default priorityClass

**The following information will help us better understand what's going on**:

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 

_If you are using earlier versions:_  
Please provide the output of the following commands (Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)
- `kubectl logs deployment/velero -n velero`
- `velero backup describe <backupname>` or `kubectl get backup/<backupname> -n velero -o yaml`
- `velero backup logs <backupname>`
- `velero restore describe <restorename>` or `kubectl get restore/<restorename> -n velero -o yaml`
- `velero restore logs <restorename>`

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`):  v1.9.0
- Velero features (use `velero client config get features`): Not Set 
- Kubernetes version (use `kubectl version`): Server Version Info{Major: "1", Minor: "22", GitVersion: ""v1.22.1"}
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by cleverhu** [comment]
I submitted a PR to fix this.

**Comment by sseago** [comment]
First, to clarify the error you got -- the priorityClass you're trying to restore that fails -- is there already a priorityClass in the cluster with the same name? If so, then this is a case where Velero should normally just skip over it, as the resource is already in the cluster (warning, but not erroring out, if the in-cluster version is not identical to the backup verson). If that's true, then we already have a fix merged from https://github.com/vmware-tanzu/velero/pull/5239 -- which will be in 1.10 and is also targeted for 1.9.2.

The issue is that Velero uses the error return (looking for already exists error) to determine whether the reason for failure is that the resource is already in the cluster, and any other error type would normally indicate a different failure that should fail a restore. The problem is that in some cases, there is other ingress validation which is failing before checking whether the resource already exists, so Velero is working with incomplete information when determining whether to error out or not. The PR linked above fixes this for all resource types, not just priority classes.

**Comment by cleverhu** [comment]
> First, to clarify the error you got -- the priorityClass you're trying to restore that fails -- is there already a priorityClass in the cluster with the same name? If so, then this is a case where Velero should normally just skip over it, as the resource is already in the cluster (warning, but not erroring out, if the in-cluster version is not identical to the backup verson). If that's true, then we already have a fix merged from #5239 -- which will be in 1.10 and is also targeted for 1.9.2.
> 
> 首先，为了澄清您得到的错误——您试图恢复的 priorityClass 失败了——集群中是否已经有一个具有相同名称的 priorityClass？如果是这样，那么在这种情况下，Velero 通常应该跳过它，因为资源已经在集群中了(如果集群内的版本与备份版本不相同，则警告，但不会出错)。如果这是真的，那么我们已经从 # 5239合并了一个修复程序——它将在1.10中，目标也是1.9.2。
> 
> The issue is that Velero uses the error return (looking for already exists error) to determine whether the reason for failure is that the resource is already in the cluster, and any other error type would normally indicate a different failure that should fail a restore. The problem is that in some cases, there is other ingress validation which is failing before checking whether the resource already exists, so Velero is working with incomplete information when determining whether to error out or not. The PR linked above fixes this for all resource types, not just priority classes.
> 
> 问题在于 Velero 使用错误返回(查找已经存在的错误)来确定失败的原因是否是资源已经在集群中，并且任何其他错误类型通常会指示应该失败恢复的不同失败。问题在于，在某些情况下，在检查资源是否已经存在之前，还有其他入口验证失败，所以 Velero 在决定是否错误输出时使用的是不完整的信息。上面链接的 PR 为所有资源类型修复了这个问题，而不仅仅是优先级类。

Yes, you're right. The code was OK before. I made some optimization, so it may be better understood. Do you have some time to review? Thank you.
